### PR TITLE
TileSet 255 and above bridge repair fix

### PR DESCRIPTION
**Pull request by E1Elite**
To do: To include .cpp in vcxproj file. If needed .inj file entries for hooks.

### CODE REVIEW

File: src/Misc/tileset255_bridgerepairfix.cpp [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+	bridges become non-repairable. The variable values of BridgeTopRight1 and BridgeBottomLeft2
+	that are parsed into, gets corrupted. To fix it, this patch allocates new addresses for those variables.
+*/
+
+#include <Helpers/Macro.h>
+
+int BridgeTopRight_1 = -1;
+int BridgeBottomLeft_2 = -1;
+
+DEFINE_HOOK(545BBA, tileset255_01, 5)
```
Comment (Metadorius) [inline src/Misc/tileset255_bridgerepairfix.cpp:14]: 1. My recommendation on how to write hook names is `[ClassName]_[MethodName]_[HookName]` or `[FunctionName]_[HookName]`.
2. Hook size is only used when you use `return 0;` so it's better to set size to `0` if you don't use it.

**Review by Metadorius — CHANGES_REQUESTED** [review]
We don't use inj file in Phobos, so you only need to include it in vcxproj and then everyone can try the nightly DLL build. I've left a couple of comments in code.

**Comment by E1Elite** [comment]
Updated, check if its fine now.

**Comment by Metadorius** [comment]
Still not included in `.vcxproj`, can't push a fix to that because your PR doesn't allow edits by maintainers.

**Comment by E1Elite** [comment]
Allow edits by maintainers is enabled from the beginning, I unchecked and checked the option again now. Latest clone or download doesn't compile straightaway for some reason, gives include errors. I compiled and tested with 2021-01-25 build, so I am avoiding change to .vcxprj file from latest commit.

Having .inj file has advantages for the modders, they can disable some hooks if they have trouble or don't want a specific change wothout going for a build from the source.

**Comment by Metadorius** [comment]
Strange, I can't push to the branch anyways.

Latest `develop` compiles fine already. Perhaps you could rebase, start a new branch and reset `develop` to `upstream/develop` ? Something like `feature/bridge-repair-fix`.

Oh and also I highly suggest hopping in on Discord to stay in touch.

**Comment by E1Elite** [comment]
May be problem could be me being on VS2019. Problem in compiling starts from 2021-02-12 commit onwards giving errors on selection.cpp and tacticalclass.h.

If you want to, go ahead and add the one line inclusion of the .cpp outside this PR.

**Comment by Metadorius** [comment]
What's the problem? I can compile just fine, are you sure YRpp submodule has the correct branch checked out?

**Comment by E1Elite** [comment]
Updated vcxprj. Problem was with YRpp branch, was using master. It would be better if YRpp URL in readme leads to correct branch.

**Comment by E1Elite** [comment]
Updated with better fix.

**Pull request merged**